nwbinspector/nwbinspector.py

```diff
@@ -292,28 +309,39 @@ def inspect_all(
         importlib.import_module(module)
     # Filtering of checks should apply after external modules are imported, in case those modules have their own checks
     checks = configure_checks(config=config, ignore=ignore, select=select, importance_threshold=importance_threshold)
-    if n_jobs != 1:
 
-        # max_workers for threading is a different concept to number of processes; from the documentation
-        # https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor
-        # we can multiply the specified number of jobs by 5
-        if n_jobs != -1:
-            max_workers = n_jobs * 5
-        else:
-            max_workers = None  # concurrents doesn't have a -1 flag like joblib; set to None to achieve this
-        with ThreadPoolExecutor(max_workers=max_workers) as executor:
-            futures = []
+    if n_jobs != 1:
+        progress_bar_options.update(total=len(nwbfiles))
+        futures = []
+        n_jobs = None if n_jobs == -1 else n_jobs  # concurrents uses None instead of -1 for 'auto' mode
+        with ProcessPoolExecutor(max_workers=n_jobs) as executor:
             for nwbfile_path in nwbfiles:
-                futures.append(executor.submit(inspect_nwb, nwbfile_path=nwbfile_path, checks=checks))
-            for future in as_completed(futures):
-                for message in future.result():
-                    yield message
+                futures.append(
+                    executor.submit(
+                        _pickle_inspect_nwb, nwbfile_path=nwbfile_path, checks=checks, skip_validate=skip_validate
+                    )
+                )
+            if progress_bar:
+                for future in tqdm(as_completed(futures), **progress_bar_options):
+                    for message in future.result():
+                        yield message
+            else:
+                for future in as_completed(futures):
+                    for message in future.result():
+                        yield message
```

Comment:
```suggestion
            completed_futures = as_completed(futures)
            if progress_bar:
                completed_futures = tqdm(completed_futures, **progress_bar_options)
            for future in completed_futures:
                for message in future.result():
                    yield message
```

Reply:
Ah yes, thought about this but wasn't sure if calling `as_completed()` acted lazily; tried it ought and it seems to.